File: src/test/java/org/kowboy/Sandbox.java
package org.kowboy;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.assertEquals;

public class Sandbox {

    @Test
    public void sandbox() {
        String[] original = new String[] {"one", "two"};

        String[] shifted = Arrays.copyOfRange(original, 1, original.length);
        assertEquals(1, shifted.length);
        assertEquals("two", shifted[0]);

        shifted = Arrays.copyOfRange(shifted, 1, shifted.length);
        assertEquals(0, shifted.length);
    }
}

File: src/test/kotlin/org/kowboy/Sandbox.kt
package org.kowboy

import org.junit.Assert
import org.junit.Test
import java.util.*

class Sandbox {
    @Test
    fun sandbox() {
        val original = arrayOf("one", "two")
        var shifted = Arrays.copyOfRange(original, 1, original.size)
        Assert.assertEquals(1, shifted.size.toLong())
        Assert.assertEquals("two", shifted[0])
        shifted = Arrays.copyOfRange(shifted, 1, shifted.size)
        Assert.assertEquals(0, shifted.size.toLong())
    }
}